src/org/safs/rest/service/models/consumers/SafsrestAdapter.groovy

```diff
@@ -149,7 +149,7 @@ class SafsrestAdapter extends ClientPOJOAdapter {
         def rootUrl = "${uri.scheme}://${uri.host}:${uri.port}"
         propertyProvider.rootUrl = rootUrl
         
-        if (userid && password) {
+        if (userid && password && (propertyProvider.authToken == null)) {
```

Comment:
I have a question.
The variable 'propertyProvider' is moved out as a class field, and it will be shared by all requests (of the same session or different sessions). But how do we know it is sure that each request will use the same 'access token'? Is that possible that 'tokenProviderServiceName', 'tokenProviderAuthTokenResource', 'userName', 'trustedUser' etc. change? If they change, will the 'access token' be the same one?